internal/build/docker.go

```diff
@@ -36,6 +36,10 @@ type localDockerBuilder struct {
 	dcli *client.Client
 }
 
+type IncrementalBuilder interface {
+	Update(ctx context.Context, changedFiles []string, name string) (digest.Digest, error)
```

Comment:
is `serviceName` also unclear?

Reply:
can you give an example of what the value would be and how the caller would be expected to use it? i'm a little bit confused about how the caller is going to use this without having race conditions

Reply:
Sure, here's how I think this needs to work based on the tilt spec:

1. User calls `tilt up $service_name`
2. CLI calls `createService` RPC, daemon creates service from nothing in k8s.
3. User calls `tilt up --watch $service_name`. 
4. CLI calls some RPC which in turn tells the daemon to watch files in a directory.
5. The daemon, when files are changed, calls some function to initiate an incremental build, somewhere. Because the thing doing the incremental build, whether it's the daemon or something in k8s needs to know which service to update, the function that initiates this action needs to take some service identifier.

We decided that the service identifier would just be the name of the service function in the Tiltfile. So if you ran `tilt up --watch blorgly_backend_local` then the value of serviceName that would be passed to `Update` would be `"blorgly_backend_local"`.

Reply:
can we create a new type for serviceName, to identify what it is?

right now, createService does not take a service name. does it need to?

what happens if the CLI calls Update twice in parallel? Does the second call use the result of the first one? or do we assume that never happens?

what happens if a build fails? how does this affect the next call to Update?

Reply:
> can we create a new type for serviceName, to identify what it is?

Yes, good idea.

> right now, createService does not take a service name. does it need to?

Yes. There's another piece of work here which is keeping track of which services are running, which is a feature that is also needed for `tilt list-services`.

> what happens if the CLI calls Update twice in parallel? Does the second call use the result of the first one? or do we assume that never happens?

Undefined! But I don't think this interface precludes any sort of synchronization mechanism

> what happens if a build fails? how does this affect the next call to Update?
 
Good question! I think handling that logic would exist outside of this function though. Maybe we ignore the digest and clean it up? Maybe we do use it?

Reply:
After talking with @landism we decided it would make it more clear if we don't return a digest. This isn't actually part of our current build pipeline that is focused on image digests.

Reply:
hmmmm....i have a feeling we're going to have to add more synchronization to this function later, but let's check this in for now and i'll write tests for the cases i'm worried about once it works. lgtm!